lib/Importer/ONNXModelLoader.cpp

```diff
@@ -206,479 +206,545 @@ static llvm::Error loadTensor(const ONNX_NAMESPACE::TensorProto &in,
   return llvm::Error::success();
 }
 
-llvm::Error ONNXModelLoader::loadOperator(const ONNX_NAMESPACE::NodeProto &op) {
-  ArgumentDictionaryTy dict = loadArgumentMap(op);
-  const std::string &typeName = op.op_type();
-
-  // Check if operator is supported in parent class, CommonOperatorLoader.
-  bool tryLoadCommonOperatorResult;
-  ASSIGN_VALUE_OR_RETURN_ERR(tryLoadCommonOperatorResult,
-                             tryLoadCommonOperator(typeName, op, dict));
-  if (tryLoadCommonOperatorResult) {
+llvm::Error ONNXModelLoader::loadConstant(const ONNX_NAMESPACE::NodeProto &op,
+                                          const ArgumentDictionaryTy &dict) {
+  /*
+    output: "Parameter6"
+    name: "Parameter6"
+    op_type: "Constant"
+    attribute {
+      name: "value"
+      t {
+        dims: 8
+        data_type: FLOAT
+        float_data: -0.161539719
+        float_data: -0.433835655
+        float_data: 0.091641359
+        float_data: -0.0168522168
+        float_data: -0.0650264397
+        float_data: -0.131737873
+        float_data: 0.0204175506
+        float_data: -0.121110231
+      }
+      type: TENSOR
+    }
+    doc_string: ""
+    domain: ""
+  */
+
+  const auto &name = op.output(0);
+  // If the tensor is pre-populated by the user of this class then we don't
+  // need to allocate a new tensor.
+  if (tensors_.count(name)) {
     return llvm::Error::success();
   }
 
-  const std::string &opName = loadOperatorName(op);
+  RETURN_ERR_IF_NOT(dict.at("value")->type() ==
+                        ONNX_NAMESPACE::AttributeProto::TENSOR,
+                    "Only Tensor type constants are supported.");
 
-  // Load tensors with values:
-  if (typeName == "Constant") {
-    /*
-      output: "Parameter6"
-      name: "Parameter6"
-      op_type: "Constant"
-      attribute {
-        name: "value"
-        t {
-          dims: 8
-          data_type: FLOAT
-          float_data: -0.161539719
-          float_data: -0.433835655
-          float_data: 0.091641359
-          float_data: -0.0168522168
-          float_data: -0.0650264397
-          float_data: -0.131737873
-          float_data: 0.0204175506
-          float_data: -0.121110231
-        }
-        type: TENSOR
-      }
-      doc_string: ""
-      domain: ""
-    */
-
-    const auto &name = op.output(0);
-    // If the tensor is pre-populated by the user of this class then we don't
-    // need to allocate a new tensor.
-    if (tensors_.count(name)) {
-      return llvm::Error::success();
-    }
+  auto *T = new Tensor();
+  RETURN_IF_ERR(loadTensor(dict.at("value")->t(), T));
```

Comment:
in case of returning here how do we free the memory for `auto *T`?

Reply:
What don't we use RAII instead of raw pointers?

Reply:
@rdzhabarov great catch thank you, I'll fix this here
@yinghai Yeah I've also been wondering this as well. I think we ought to be using unique_ptrs here and I don't see any reason so far not to. @rdzhabarov thoughts on this?

Reply:
#2115

Reply:
Agree, it's better to have unique ptr to control the life time of that pointer. Can be done separately from this change.